app/scripts/pdf.directive.js

```diff
@@ -61,11 +64,11 @@ angular.module('dashboard')
 
                 scope.$watch(
                     function () {
-                        return { hide: scope.hide };
+                        return { modalActive: $rootScope.modalActive };
                     },
                     function (data) {
                         if (angular.isObject(data)) {
```

Comment:
The watch examples suggest it might be good to check if value has changed, there might be calls to watch even when it isn't.